### PR TITLE
Fix swift4 warnings

### PR DESCRIPTION
**Pull request by Nekitosss**

### CODE REVIEW

File: Sources/Core/Private/Helpers.swift [hunk 1]
```diff
@@ -29,10 +29,6 @@ protocol TypeGetter {
 
 protocol SwiftTypeGetter: TypeGetter {}
 
-extension ImplicitlyUnwrappedOptional: SwiftTypeGetter {
```
Comment (ivlevAstef) [inline Sources/Core/Private/Helpers.swift]: @available(swift, obsoleted: 4.1)

File: Sources/Core/Private/Resolver.swift [hunk 2]
```diff
@@ -150,7 +150,7 @@ class Resolver {
     if hasMany {
       let filterComponents = components.filter{ !stack.contains($0.info) } // Remove objects contains in stack
       assert(nil == object, "Many injection not supported")
-      return filterComponents.flatMap{ makeObject(by: $0, use: nil) }
+		return filterComponents.compactMap{ makeObject(by: $0, use: nil) }
```
Comment (ivlevAstef) [inline Sources/Core/Private/Resolver.swift]: #if swift(>= 4.1)
#else
#endif
maybe for ImplicitlyUnwrappedOptional need use 'if'

File: Sources/Scan/DIScan.swift [hunk 3]
```diff
@@ -14,12 +14,12 @@ public protocol DIScanned {}
 open class DIScan {  
   static func types(_ valid: (AnyClass)->Bool, _ predicate: (AnyClass)->Bool, _ bundle: Bundle?) -> [AnyClass] {
     if let bpath = bundle?.bundlePath {
-      return scanned.flatMap { cls, clsBPath in
+      return scanned.compactMap { cls, clsBPath in
```
Comment (ivlevAstef) [inline Sources/Scan/DIScan.swift]: #if swift(>= 4.1)
#else
#endif
or for simple change:
```
#if swift(>= 4.1)
#else
extension Sequence
{
  func compactMap(...) { ... )
}
#endif
```

Comment (Nekitosss) [reply Sources/Scan/DIScan.swift]: Fixed

File: Sources/Core/Private/Resolver.swift (hunk 2 above)
Comment (Nekitosss) [reply Sources/Core/Private/Resolver.swift]: Fixed

**Pull request merged**